ibm_pathfinder_sdk/eventpublisher.py

```diff
@@ -28,46 +28,76 @@ class NULL_NAMESPACE:
-                    self.connectorState["delete"].remove({"schema":event.payload.json_schema_ref,"from_edf_id":event.payload.from_edf_id,"to_edf_id":event.payload.to_edf_id})
-            return 0    
-
-
+                # upsert update state only
+                if ( event.event_type == "upsert"):
+                    if event.payload.json_schema_ref not in self.connectorState["state"]:
+                        self.connectorState["state"][event.payload.json_schema_ref] = {} 
+                    self.connectorState["state"][event.payload.json_schema_ref][event.payload.edf_id] = str(uuid.uuid3(NULL_NAMESPACE, event.payload.toJSON()))
+                # delete update state and delete/last run data
+                if ( event.event_type == "delete"):
+                    if ( event.payload.json_schema_ref in self.connectorState["state"] and 
+                         event.payload.edf_id in self.connectorState["state"][event.payload.json_schema_ref] ):
+                        del self.connectorState["state"][event.payload.json_schema_ref][event.payload.edf_id]
+                    if ( event.payload.json_schema_ref in self.connectorState["delete"] and 
+                         event.payload.edf_id in self.connectorState["delete"][event.payload.json_schema_ref] ):
+                        del self.connectorState["delete"][event.payload.json_schema_ref][event.payload.edf_id]
+            # state update for System entity
+            if isinstance(event.payload.__class__.__base__(), pfmodelclasses.SystemRelationship):
+                # upsert update state only
+                if ( event.event_type == "upsert"):
+                    if event.payload.json_schema_ref not in self.connectorState["state"]:
+                        self.connectorState["state"][event.payload.json_schema_ref] = {} 
+                    self.connectorState["state"][event.payload.json_schema_ref][event.payload.from_edf_id + "|" + event.payload.to_edf_id] = str(uuid.uuid3(NULL_NAMESPACE, event.payload.toJSON()))
+                # delete update state and delete/last run data
+                if ( event.event_type == "delete"):
+                    if ( event.payload.json_schema_ref in self.connectorState["state"] and 
+                         event.payload.edf_id in self.connectorState["state"][event.payload.json_schema_ref] ):
+                        del self.connectorState["state"][event.payload.json_schema_ref][event.payload.from_edf_id + "|" + event.payload.to_edf_id]   
+                    if ( event.payload.json_schema_ref in self.connectorState["delete"] and 
+                         event.payload.edf_id in self.connectorState["delete"][event.payload.json_schema_ref] ):
+                        del self.connectorState["delete"][event.payload.json_schema_ref][event.payload.from_edf_id + "|" + event.payload.to_edf_id] 
+            return 0 
+
```

Comment:
Hi Peter, this looks good, but I would consolidate these two "if" statements for Entity and Relationship into a single block, where you just use that "if" statement to obtain the eventId. That would make the code shorter and clearer (unless I'm missing something).